pkg/types/validation/installconfig.go

```diff
@@ -1589,8 +1589,32 @@ func validateFencingCredentials(installConfig *types.InstallConfig) (errors fiel
 	allErrs := field.ErrorList{}
 	if fencingCredentials != nil {
 		allErrs = append(allErrs, common.ValidateUniqueAndRequiredFields(fencingCredentials.Credentials, fldPath, func([]byte) bool { return false }, "credentials")...)
+		allErrs = append(allErrs, validateFencingForPlatform(installConfig, fldPath)...)
+		allErrs = append(allErrs, validateFencingAndHostsMutuallyExclusive(installConfig, fldPath)...)
 	}
 	allErrs = append(allErrs, validateCredentialsNumber(installConfig.ControlPlane, fencingCredentials, fldPath.Child("credentials"))...)
 
 	return allErrs
 }
+
+func validateFencingForPlatform(config *types.InstallConfig, fldPath *field.Path) field.ErrorList {
+	errs := field.ErrorList{}
+	switch {
+	case config.None != nil:
+	case config.BareMetal != nil:
+		if len(config.Platform.BareMetal.Hosts) > 0 {
+			errs = append(errs, field.Forbidden(fldPath, "fencing is mutually exclusive with hosts, please remove either of them"))
+		}
+	default:
+		errs = append(errs, field.Forbidden(fldPath, fmt.Sprintf("fencing is only supported on baremetal or none platforms, instead %s platform was found", config.Platform.Name())))
+	}
+	return errs
+}
+
+func validateFencingAndHostsMutuallyExclusive(config *types.InstallConfig, fldPath *field.Path) field.ErrorList {
```

Comment:
Since we merge this check into `validateFencingForPlatform`, we should remove this function and its calls?

Sorry, one last comment above 😅

Reply:
No worries, you are absolutely right - I shouldn't have missed that :see_no_evil: